packages/react-table/src/components/Table/utils/decorators/collapsible.tsx

```diff
@@ -56,12 +56,51 @@ export const expandedRow = (colSpan: number) => {
         extraParams: { contentId = 'expanded-content' }
       }
     }: IExtra
-  ): decoratorReturnType =>
-    rowData.hasOwnProperty('parent') && {
-      // todo: rewrite this logic, it is not type safe
-      colSpan: colSpan + (!!rowData.fullWidth as any),
-      id: contentId + rowIndex,
-      className: rowData.noPadding && css(styles.modifiers.noPadding)
-    };
+  ): decoratorReturnType | false => {
+    if (rowData.hasOwnProperty('parent')) {
+      const formatter: decoratorReturnType = {
+        id: contentId + rowIndex,
+        className: rowData.noPadding && css(styles.modifiers.noPadding)
+      };
+
+      if (!rowData.useAllCellInExpandedContent) {
+        formatter.colSpan = colSpan + (rowData.fullWidth ? 1 : 0);
+      }
+
+      return formatter;
+    }
+
+    return false;
+  };
   return expandedRowFormatter;
 };
+
+export const skipRemainingExpandedRow = () => {
+  const skipRemainingExpandedRowFormatter = (
+    value: IFormatterValueType,
+    {
+      rowIndex,
+      rowData,
+      columnIndex,
+      column: {
+        extraParams: { contentId = 'expanded-content' }
+      }
+    }: IExtra
+  ): decoratorReturnType | false => {
+    if (rowData.hasOwnProperty('parent')) {
+      const formatter: decoratorReturnType = {
+        id: contentId + rowIndex + '-' + columnIndex,
+        className: rowData.noPadding && css(styles.modifiers.noPadding)
+      };
+
+      if (!rowData.useAllCellInExpandedContent) {
+        formatter.skipCell = true;
```

Comment:
Tried using `isVisible: false`, but it always ended up being `true`. Not sure if is the way the transformation work, or if the `mapProps` (IIRC that's the name of the last transformation) does that. But i didn't set any `isVisible: true` on my code.